test/objects.coffee

```diff
@@ -439,6 +439,12 @@ test "#3216: For loop declaration as a value of an implicit object", ->
   arrayEq ob.b, test
   arrayEq ob.c, test
   arrayEq ob.d, test
+  byFirstKey =
```

Comment:
added a couple tests since the `@insideForDeclaration`/#3216 stuff really only kicks in on the first line of an implicit object